### ISSUE TITLE
[Big Picture Mode] Heavy frame drop/slow performance on Collections Page

### ISSUE THREAD

**Issue by deanrih** [ValveSoftware/steam-for-linux#9834]
#### Your system information

* Steam client version (build number or date): `1689034492`
* Distribution (e.g. Ubuntu): Manjaro (Arch)
* Opted into Steam client beta?: [Yes/No] Yes
* Have you checked for system updates?: [Yes/No] Yes
* Steam Logs: [steam-logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/12024538/steam-logs.tar.gz)
* GPU: NVidia 2060

#### Please describe your issue in as much detail as possible:
On Big Picture Mode, when seeing or changing to Collections page, the whole app (Big Picture Mode) becomes very slow compared to other page/tab, even after the page completed the transition, selecting the page's content still slow.

Here's the screen capture showing the problem:
https://streamable.com/33gsq8

#### Steps for reproducing this issue:

1. Make sure you have many collections containing many games, probably 5 games per collection is enough
2. Go to Big Picture Mode
3. Switch to the Collections Page

**Comment by ranplayer** [comment]
This issue is also present for AMD GPUs. For 4K resolution it's even worse.

My specs:
- Steam version: 1689034492
- OS: Arch
- Kernel: 6.4.3
- GPU: RX 6700 XT (Mesa 23.1.3)
- CPU: Ryzen 7 5700 G

**Comment by hustin-paul** [comment]
Also present for me.

Arch, 6.4.3 kernel, 4090 535.54.03

**Comment by kisak-valve** [comment]
Hello, per "Improved library performance for users with large numbers of collections." in the [2023-07-28](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/3662040707264180330) Steam client beta update, please retest this issue.

**Comment by hustin-paul** [comment]
@kisak-valve Latest steam beta (2023-07-28) there is no improvement for me. Same results.

**Comment by deanrih** [comment]
> Hello, per "Improved library performance for users with large numbers of collections." in the [2023-07-28](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/3662040707264180330) Steam client beta update, please retest this issue.

I updated to version `1690583737` and it seems the issue still persists and without any improvement. Also, I should add in the detail that I have around 5 dynamic collections; that may help to track down the problem.

**Comment by deathxxx123** [comment]
Here too. Super slow performance witn Big picture UI:
Pop!_OS 22.04
Nvidia 535.113.01
4K display

**Comment by rhoot** [comment]
You might be having the same performance issue I do (#10084). You could try the workaround in that issue, but it needs to be reapplied every time the steam client updates.

**Comment by radgeRayden** [comment]
Can confirm this is still the case. Nvidia on a 4K display.

**Comment by Sturmlocke86** [comment]
Performance with the New Big Picture mode on Nvidia is still an issue for me too:

OpenSUSE Leap 15.5
Intel® Core™ i7-7700 CPU @ 3.60GHz × 8
NVIDIA RTX A2000/PCIe/SSE2
Gnome 41.8
X11
545.29.06

Tried both windowed and fullscreen mode, performance is aweful in both cases.

**Comment by deathxxx123** [comment]
Now fixed itself, the UI is very good performance, but sometimes when restart steam will bring back extremely slow FPS. After restart again, fixed again.
I use Steam bela(with native lib's replacement), Manjaro Gnome, lastest Nvidia driver and RTX 3060 Ti 8GB

**Comment by mambucodev** [comment]
I'm reporting this issue as well

Arch Linux 6.10.9-arch1-2
Intel(R) Core(TM) i5-10400F
Gigabyte AMD Radeon RX 5700 XT
Gnome 46.4
Wayland 1.23.1-1

Tried both steam and steam beta, still getting lag.

**Comment by frantic1048** [comment]
Encountered the same issue on latest steam client:
- The normal UI mode performed well
- In Big Picture Mode, it shows everything in very low FPS. (e.g. moving focus from one to another item could take more than 1 second for UI update)

Environment info:
- Steam version: `1726256783`
- OS: Arch Linux x86_64
- Kernel: Linux 6.10.8-arch1-1
- WM: i3 (X11)
- CPU: AMD Ryzen Threadripper 2950X
- GPU: AMD Radeon RX 6800 XT

**Comment by theLucasAntunes** [comment]
Same issue for me!

Tried in both Steam Official Release and Beta

* Steam Version:  1726683985
* OS: Windows 11
* CPU: AMD Ryzen 9 5950X
* GPU: EVGA GTX 3070 TI FTW3

**Comment by KingKusuo** [comment]
>     * GPU: EVGA GTX 3070 TI FTW3
wait, GTX 3070?

**Comment by mambucodev** [comment]
I tried big picture on a brand new install and it's not lagging. it lagged in my old pc, where I had like 3k packages installed. could this be related?

**Comment by deanrih** [comment]
> I tried big picture on a brand new install and it's not lagging. it lagged in my old pc, where I had like 3k packages installed. could this be related?

Packages count shouldn't affecting BPM performance... however, installed packages might have different version between your new install and old install, maybe you could show us which packages that exists on both installs that have different version?

**Comment by mambucodev** [comment]
Hey, updates here:
- I installed arch linux on a mini-pc that has an intel celeron CPU, and big picture was really fluid. No lag, no flashing.
- I reinstalled arch linux in my main desktop, with i5 10400F and RX 5700 XT, and big picture is still really, really, laggy.

**Comment by gentuser** [comment]
how I solved in KDE:
in /usr/lib/steam/bin_steam.sh  I changed the line 
`from exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@"  `
to 
`exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@"  --use-gl=desktop `

Randomly the big-picture still presents lags, so I had to tick off "Allow applications to block compositing". I know that this feature is needed in some games. So keep it on in case any game crashes or has performance issues.

**Comment by kgagnon** [comment]
Hate to be that guy that comes in and says "same" but... same.

Fedora 41 / GNOME 47 / Wayland / 3440 x 1440
Linux 6.11.7-300.fc41.x86_64
AMD Radeon™ RX 7900 XTX
AMD Ryzen™ 7 3700X × 16


We're not talking FPS drop, we're not even in FPS, we're in SPF. It's that bad.

**Comment by hadley8899** [comment]
Just adding a comment to this issue to make it higher priority! Nvidia 4070ti with the 560 driver

**Comment by hardpenguin** [comment]
Is this not the same as https://github.com/ValveSoftware/steam-for-linux/issues/11255 ?

Anyway yeah I can confirm this is an issue for me as well. NVIDIA 3060Ti with the 525 line driver on Debian Sid.

**Comment by deanrih** [comment]
> Is this not the same as #11255 ?
> 
> Anyway yeah I can confirm this is an issue for me as well. NVIDIA 3060Ti with the 535 line driver on Debian Sid.

not exactly, this is specifically on the "Collection Page" where the performance goes even lower than it already is for the rest BPM, if you haven't watched the video above, that is

**Comment by deanrih** [comment]
> Is this not the same as #11255 ?
> 
> Anyway yeah I can confirm this is an issue for me as well. NVIDIA 3060Ti with the 535 line driver on Debian Sid.

that being said, I haven't tried to run it on Gamescope like in the issue you mentioned, maybe I'll try it after getting home

**Comment by hardpenguin** [comment]
> not exactly, this is specifically on the "Collection Page" where the performance goes even lower than it already is for the rest BPM, if you haven't watched the video above, that is

Yeah I get it, the issue with "Collections Page" specifically affects me as well.

**Comment by RalphB15** [comment]
I’m experiencing the same issue, but I’ve noticed that the frame drop specifically targets the BP UI elements. Interestingly, when I quickly switch trough the 'Menu-Button' on the bottom left, to the 'Store', the framerate within the Store container itself is perfectly fine.

The lag is most visible when interacting with global UI elements: hovering over the 'Menu' button triggers a highlight animation that runs at roughly 0.5 FPS (one frame every 2 seconds), while the sub-content (store) remains fluid.

**System:**
  Host: **** Kernel: 6.14.0-37-generic arch: x86_64 bits: 64
  Desktop: Cinnamon v: 6.4.8 Distro: Linux Mint 22.2 Zara
**Machine:**
  Type: Desktop Mobo: Micro-Star model: MAG B460M BAZOOKA (MS-7C83) v: 2.0
    serial: <superuser required> UEFI: American Megatrends v: A.00
    date: 05/13/2020
**CPU:**
  Info: 6-core model: Intel Core i5-10600K bits: 64 type: MT MCP cache:
    L2: 1.5 MiB
  Speed (MHz): avg: 2425 min/max: 800/4800 cores: 1: 4718 2: 800 3: 800
    4: 800 5: 800 6: 800 7: 4698 8: 4685 9: 800 10: 4700 11: 800 12: 4701
**Graphics:**
  Device-1: NVIDIA GA104 [GeForce RTX 3070] driver: nvidia v: 580.95.05
  Display: x11 server: X.Org v: 21.1.11 with: Xwayland v: 23.2.6 driver: X:
    loaded: nvidia unloaded: fbdev,modesetting,nouveau,vesa
    gpu: nv_platform,nvidia,nvidia-nvswitch resolution: 1: 2732x1536~60Hz
    2: 5120x2880~60Hz
  API: EGL v: 1.5 drivers: nouveau,nvidia,swrast
    platforms: gbm,x11,surfaceless,device
  API: OpenGL v: 4.6.0 compat-v: 4.5 vendor: nvidia mesa v: 580.95.05
    renderer: NVIDIA GeForce RTX 3070/PCIe/SSE2
  API: Vulkan v: 1.3.275 drivers: N/A surfaces: xcb,xlib



**Edit 1:**
Enabling "Enable GPU accelerated rendering in web views" somehow fixed it for me...

**Edit 2:** (Noticed very stuttery video playback)
Enabling GPU acceleration causes Steam to seize exclusive control of the X11 rendering queue and VSync locks, effectively starving my browser and desktop compositor of GPU cycles. Because the NVIDIA driver and Cinnamon's compositor (Muffin) cannot efficiently arbitrate between two high-priority "browser" processes, the system forces my other windows into a "wait state" while Steam is active. This results in the fluid Steam UI, but at the direct expense of every other application's framerate.

Closing the steam window (minimizing to tray) or reverting the changes of Edit 1 fixes the stuttering. I guess I'll have to pick my poison for now.
